hazelcast/src/test/java/com/hazelcast/config/CacheSimpleConfigTest.java

```diff
@@ -0,0 +1,60 @@
+package com.hazelcast.config;
+
+import com.hazelcast.test.HazelcastParallelClassRunner;
+import com.hazelcast.test.HazelcastTestSupport;
+import com.hazelcast.test.annotation.ParallelTest;
+import com.hazelcast.test.annotation.QuickTest;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+
+import static org.junit.Assert.fail;
+
+@RunWith(HazelcastParallelClassRunner.class)
+@Category({QuickTest.class, ParallelTest.class})
+public class CacheSimpleConfigTest extends HazelcastTestSupport {
+
+    @Test
+    public void givenCacheLoaderIsConfigured_whenConfigureCacheLoaderFactory_thenThrowIllegalStateException() {
+        CacheSimpleConfig config = new CacheSimpleConfig();
+        config.setCacheLoader("foo");
+
+        try {
+            config.setCacheLoaderFactory("bar");
+            fail("Should throw IllegalStateException");
```

Comment:
just a matter of style, you could also use a `@Rule ExpectedException` and rewrite as:
```
expectedException.expect(IllegalStateException.class);
config.setCacheLoaderFactory("bar");
```
IMHO makes the expected outcome more clear